src/v/cluster/self_test_backend.cc

```diff
@@ -28,36 +28,45 @@ self_test_backend::self_test_backend(
   model::node_id self,
   ss::sharded<node::local_monitor>& nlm,
   ss::sharded<rpc::connection_cache>& connections,
+  ss::sharded<cloud_storage::remote>& cloud_storage_api,
   ss::scheduling_group sg)
   : _self(self)
   , _st_sg(sg)
   , _disk_test(nlm)
-  , _network_test(self, connections) {}
+  , _network_test(self, connections)
+  , _cloud_test(cloud_storage_api) {}
 
 ss::future<> self_test_backend::start() {
     co_await _disk_test.start();
     co_await _network_test.start();
+    co_await _cloud_test.start();
 }
 
 ss::future<> self_test_backend::stop() {
     auto f = _gate.close();
     co_await _disk_test.stop();
     co_await _network_test.stop();
+    co_await _cloud_test.stop();
     co_await _lock.get_units(); /// Ensure outstanding work is completed
     co_await std::move(f);
 }
 
 ss::future<std::vector<self_test_result>> self_test_backend::do_start_test(
-  std::vector<diskcheck_opts> dtos, std::vector<netcheck_opts> ntos) {
+  std::vector<diskcheck_opts> dtos,
+  std::vector<netcheck_opts> ntos,
+  std::vector<cloudcheck_opts> ctos) {
     auto gate_holder = _gate.hold();
     std::vector<self_test_result> results;
+
     for (auto& dto : dtos) {
         try {
             dto.sg = _st_sg;
             if (!_cancelling) {
-                auto dtr = co_await _disk_test.run(dto).then(
-                  [](auto result) { return result; });
-                std::copy(dtr.begin(), dtr.end(), std::back_inserter(results));
+                auto dtr = co_await _disk_test.run(dto);
+                results.insert(
+                  results.end(),
+                  std::make_move_iterator(dtr.begin()),
+                  std::make_move_iterator(dtr.end()));
```

Comment:
> rpk: add cloud storage self-test
Add rpk bindings for cluster's cloud storage self test, and related
cluster rpc/frontend/backend code.

This commit is about rpk, but mentions "related cluster rpc/frontend/backend code". This later part, these C++ changes that don't seem to be related to rpk, should be in a different (or previous) commit where the changes can be explained in the commit message.